src/charm.py

```diff
@@ -239,30 +245,24 @@ def is_kubeconfig_available(self):
                     return True
         return False
 
-    def set_service_cidr(self, event):
-        """Repeat received service-cidr from k8s-cp to each kube-ovn unit.
+    def set_peer_data(self, event, cni_data_key, peer_relation_data_key=""):
+        """Repeat received CNI relation data to each kube-ovn unit.
+        The keys for the CNI relation data and peer relation data
+        are assumed to be the same unless otherwise specified.
 
-        service-cidr is received over the cni relation only from
+        CNI relation data is received over the cni relation only from
         kubernetes-control-plane units.  the kube-ovn peer relation
         shares the value around to each kube-ovn unit.
         """
-        cni_service_cidr = event.relation.data[event.unit].get("service-cidr")
-        if cni_service_cidr:
+        if not peer_relation_data_key:
+            peer_relation_data_key = cni_data_key
+        cni_data = event.relation.data[event.unit].get(cni_data_key)
+        if cni_data:
             for relation in self.model.relations["kube-ovn"]:
-                relation.data[self.unit]["service-cidr"] = cni_service_cidr
+                relation.data[self.unit][peer_relation_data_key] = cni_data
```

Comment:
this method, i understand the reasoning behind it.  It prevents two copies of what's effectively "repeat this data blob across the kube-ovn peers

what if you instead just wrote a method:

```python
def cni_to_kube_ovn(event):
   for key in ["service-cidr", "image-registry"]:
      cni_data = event.relation.data[event.unit].get(key)
      if not cni_data:
          continue
      for relation in self.model.relations["kube-ovn"]:
          relation.data[self.unit][key] = cni_data
```

and later on call it with:

```python 
    def on_cni_relation_changed(self, event):
        self.cni_to_kube_ovn(event)
        ...
```